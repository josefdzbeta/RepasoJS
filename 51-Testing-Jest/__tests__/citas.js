/**
 * @jest-environment jsdom
 */

import  Citas from "../js/classes/Citas";

describe('probar la clase de citas', ()=>{

    const citas = new Citas();
    const id = Date.now();

    test('Agregar una nueva cita', ()=>{
        const citaObj = {
            id,
            mascota: 'Hook',
            propietario: 'Pedro',
            telefono: '123123',
            fecha: '10122023',
            hora:'10:30',
            sintomas: 'No para de comer'
        }
        citaObj.id = Date.now();
        citas.agregarCita(citaObj);

        //Prueba
        expect(citas).toMatchSnapshot();
        
       
    });
    test('Actualizar cita', ()=>{
        const citaActualizada = {
            id,
            mascota: 'Hook',
            propietario: 'Pedro',
            telefono: '123123',
            fecha: '101220232',
            hora:'11:30',
            sintomas: 'No para de dormir'
        };
        citas.editarCita(citaActualizada);
        expect(citas).toMatchSnapshot();
    });
    test('Eliminar Cita', ()=>{
        citas.eliminarCita(id);
        expect(citas).toMatchSnapshot();
    })
});