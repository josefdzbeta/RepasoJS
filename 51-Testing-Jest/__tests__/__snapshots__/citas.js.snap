// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probar la clase de citas Actualizar cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10122023",
      "hora": "10:30",
      "id": 1670709714130,
      "mascota": "Hook",
      "propietario": "Pedro",
      "sintomas": "No para de comer",
      "telefono": "123123",
    },
  ],
}
`;

exports[`probar la clase de citas Agregar una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10122023",
      "hora": "10:30",
      "id": 1670709714130,
      "mascota": "Hook",
      "propietario": "Pedro",
      "sintomas": "No para de comer",
      "telefono": "123123",
    },
  ],
}
`;

exports[`probar la clase de citas Eliminar Cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10122023",
      "hora": "10:30",
      "id": 1670709714130,
      "mascota": "Hook",
      "propietario": "Pedro",
      "sintomas": "No para de comer",
      "telefono": "123123",
    },
  ],
}
`;
